include/match.h

```diff
@@ -0,0 +1,153 @@
+#ifndef INCLUDE_MATCH_H_
+#define INCLUDE_MATCH_H_
```

Comment:
The variable naming scheme here is not standardized across source files. Please use `_FULLTEXT_NAME_H_` across the board (that is, this would be `_FULLTEXT_MATCH_H_`).